### PR TITLE
Development

### PR DESCRIPTION
**Pull request by szczepanskikrs**
### Description

Intergation tests added.

### List general changes for this pull request:

* Added integration tests
* Added Awaitility
* Reworked server builder so it will not chain in incorrect manner
* Added missing JavaDocs 

### List of modules this pull request affects:
* Server

### CODE REVIEW

File: client/src/main/java/battleships/game/OpponentBoardViewController.java [hunk 1]
```diff
@@ -92,13 +94,13 @@ private void processShot(Integer fieldPosition) {
   }
 
   @Override
-  public void accept(Messageable event) {
-    if (event instanceof SalvoCount) {
+  public void accept(Messageable data) {
+    if (data instanceof SalvoCount) {
```
Comment (M-Aarsman) [inline client/src/main/java/battleships/game/OpponentBoardViewController.java]: soooo many instance of

File: client/src/main/java/battleships/game/OpponentBoardViewController.java [hunk 2]
```diff
@@ -92,13 +94,13 @@ private void processShot(Integer fieldPosition) {
...
       salvoBtn.setDisable(false);
-      processSalvoCount((SalvoCount) event);
+      processSalvoCount((SalvoCount) data);
       log.info("SalvoCount received by controller");
-    } else if (event instanceof SalvoResult) {
-      SalvoResult salvoResult = (SalvoResult) event;
+    } else if (data instanceof SalvoResult) {
```
Comment (M-Aarsman) [inline client/src/main/java/battleships/game/OpponentBoardViewController.java]: and here

File: client/src/main/java/battleships/logging/validation/Validator.java [hunk 3]
```diff
@@ -6,9 +6,11 @@
 
 import battleships.logging.ConfigValue;
 import battleships.logging.ConfigValueName;
+
 import java.util.Map;
 
 
+
```
Comment (M-Aarsman) [inline client/src/main/java/battleships/logging/validation/Validator.java]: you line new lines :D

File: client/src/test/java/battleships/game/BoardTest.java [hunk 4]
```diff
@@ -1,26 +1,93 @@
...
+            .filter(field -> field.getPosition() < 10)
+            .forEach(field -> field.setUnbrokenShipPartOn());
+    //then
+    assertThat(board.unbrokenMastCount()).isEqualTo(10);
+  }
+
+  @Test
+  public void whenPlacingFiveFieldsWithUnbrokenShipPart_andFiveFieldsWithBrokenShipPart_expectedCountOfUnbrokenMastsIs5(){
+    //when
+    board.getFields().stream()
```
Comment (M-Aarsman) [inline client/src/test/java/battleships/game/BoardTest.java]: maybe you can extract it to private method with meaningful name?

File: client/src/test/java/battleships/game/BoardTest.java [hunk 5]
```diff
@@ -1,26 +1,93 @@
...
+            .filter(field -> field.getPosition() >= 20 && field.getPosition() < 25)
+            .forEach(field -> field.setBrokenShipPartOn());
+    //then
+    assertThat(board.unbrokenMastCount()).isEqualTo(5);
+  }
+
+  @Test
+  public void whenPlacingFiveFieldsWithUnbrokenShipPart_andShootingTwoOfThem_expectedCountOfUnbrokenMastsIs3(){
+    //when
+    List<Field> fieldsWithUnbrokenShipPart
```
Comment (M-Aarsman) [inline client/src/test/java/battleships/game/BoardTest.java]: the same

File: client/src/test/java/battleships/game/BoardTest.java [hunk 6]
```diff
@@ -1,26 +1,93 @@
...
   }
 
   @Test
-  public void whenBuildingBoardWithCleanFields_expectCountOfEmptyFieldIs100() {
+  public void whenPlacingOnSomeFieldUnbrokenShipPart_expectedColorOfRectangleBindWithBoardNodeIsAsColorYellow(){
     //when
-    Board board = Board.build();
-    long sizeOfEmptyFields = board.getFields().stream().filter(Field::isEmpty).count();
+    int someFieldIndex = 0;
+    board.getFields().get(someFieldIndex).setUnbrokenShipPartOn();
```
Comment (M-Aarsman) [inline client/src/test/java/battleships/game/BoardTest.java]: this look bad get().get().set() ?

File: server/src/main/java/battleships/communication/Server.java [hunk 7]
```diff
@@ -10,21 +10,30 @@
 
 /**
  * This class represents a server that waits for connection from two clients.
- * It's main responsibility is to accepts client's connections and create a list of sockets bound to this connections.
+ * It's main responsibility is to accepts client's connections and create a list of sockets
```
Comment (M-Aarsman) [inline server/src/main/java/battleships/communication/Server.java]: single responsibility (AND)

File: server/src/test/java/battleships/integrationtests/MessagesAreSentFromServerIT.java [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.jayway.awaitility.Awaitility.await;
+import static org.assertj.core.api.Java6Assertions.assertThat;
+
+public class MessagesAreSentFromServerIT {
```
Comment (M-Aarsman) [inline server/src/test/java/battleships/integrationtests/MessagesAreSentFromServerIT.java]: javadoc

Comment (M-Aarsman) [reply server/src/test/java/battleships/integrationtests/MessagesAreSentFromServerIT.java]: ok this is test, name of a class doesn't suggest that

File: server/src/test/java/battleships/integrationtests/MessagesAreSentFromServerIT.java [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+  @BeforeMethod
+  public void setUp() {
+    isServerWorking = new AtomicBoolean(false);
+    isClientWorking = new AtomicBoolean(false);
+  }
+
+  @Test(dataProvider = "ParametrizedSalvos")
+  public void whenServerSendsSalvo_expectClientCanInterpretItCorrectly(IntegProvider integProvider) throws Exception {
+    //when
+    startServerAndClient(integProvider);
```
Comment (M-Aarsman) [inline server/src/test/java/battleships/integrationtests/MessagesAreSentFromServerIT.java]: maybe extract to method?

File: server/src/test/java/battleships/integrationtests/MessagesAreSentFromServerIT.java [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+    Salvo receivedMessage = (Salvo) clientHandler.receiveMessage();
+    //then
+    assertThat(receivedMessage.getSalvoPositions()).isEqualTo(expectedSalvo.getSalvoPositions());
+
+  }
+
+  @Test(dataProvider = "ParametrizedSalvoResults")
+  public void whenServerSendsSalvoResult_expectClientCanInterpretItCorrectly(IntegProvider integProvider) throws Exception {
+    //when
+    startServerAndClient(integProvider);
```
Comment (M-Aarsman) [inline server/src/test/java/battleships/integrationtests/MessagesAreSentFromServerIT.java]: the same

File: server/src/test/java/battleships/integrationtests/MessagesAreSentFromServerIT.java [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+    softAssert.assertEquals(receivedMessage.getGameResult(), expectedSalvoResult.getGameResult());
+    softAssert.assertEquals(receivedMessage.getResultList(), expectedSalvoResult.getResultList());
+    softAssert.assertEquals(receivedMessage.getSalvoPositions(), expectedSalvoResult.getSalvoPositions());
+    softAssert.assertAll();
+  }
+
+  @Test(dataProvider = "ParametrizedWelcomeMessages")
+  public void whenServerSendsWelcomeMessage_expectClientCanInterpretItCorrectly(IntegProvider integProvider) throws Exception {
+    //when
+    startServerAndClient(integProvider);
```
Comment (M-Aarsman) [inline server/src/test/java/battleships/integrationtests/MessagesAreSentFromServerIT.java]: and here

File: server/src/test/java/battleships/integrationtests/ReceiveMessageIT.java [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+public class ReceiveMessageIT {
+  private SoftAssert softAssert = new SoftAssert();
+
+  @Test(dataProvider = "salvoProvider")
+  public void whenSalvoMessageIsSendToServer_expectSameMessageToBeReceivedByServer(IntegProvider integProvider) throws Exception {
+    //given
+    final String LOCALHOST = "127.0.0.1";
+    Server server = createServer();
+    //when
+    runSenderThread(LOCALHOST, server.getServerSocketPort(), integProvider.messageForPlayer0());
```
Comment (M-Aarsman) [inline server/src/test/java/battleships/integrationtests/ReceiveMessageIT.java]: extract to method?

File: client/src/main/java/battleships/communication/server/ServerConnector.java [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+  }
+
+  /**
+   * Checks if connection to server is properly set up by waiting for proper message from server.
+   *
+   * @return true if connection is valid, false otherwise
+   */
+  public boolean setUp() {
+    boolean connectionEstablish = false;
+    if (serverComm.waitForMessage() instanceof WelcomeMessage) {
```
[the remark this answers is not in the document]
Comment (igasm) [reply client/src/main/java/battleships/communication/server/ServerConnector.java]: issue #108

File: client/src/main/java/battleships/game/OpponentBoardViewController.java (hunk 1 above)
Comment (igasm) [reply client/src/main/java/battleships/game/OpponentBoardViewController.java]: issue #108

File: client/src/main/java/battleships/game/OpponentBoardViewController.java (hunk 2 above)
Comment (igasm) [reply client/src/main/java/battleships/game/OpponentBoardViewController.java]: issue #108

**Pull request merged**